contracts/auction/integer-key-auction.py

```diff
@@ -80,15 +80,21 @@ def __command_auction__(state, bindings, pargs) :
         return
 
     if options.command == 'prime' :
-        bidinfo = dict(options.attestation[0])
+        assert type(options.attestation) is list
+        assert len(options.attestation) == 3
+
+        bidinfo = options.attestation[0]
         dependencies = options.attestation[1]
         signature = options.attestation[2]
```

Comment:
Curious: How are these dependencies specified. I suppose as (contract_id, statehash) pairs? I am afraid I can't find the usage.

Reply:
there are several examples in integer-key-auction.

Reply:
I am sure there was, but I need to understand this from you. I just don't see how the attestation option gets populated with the (contract_id, state hash) pairs based on past invocation responses.

Reply:
why is this question relevant for the PR?

Reply:
I was just thinking about your observation that some transactions succeed even though they are supposed to fail based on dependency checks. So was trying to see the whole trail of how dependencies propagate. Was that observation not specific to this PR?

Reply:
Separate problem. The dependencies are sent, the transaction is not committed. But a txnid is returned. This is a specific contract. If the dependencies were not correct then it should fail every time. It doesn't. Again... i think this is out of scope.

Reply:
ok thank you!